src/main/cpp/log_writer.cpp

```diff
@@ -121,10 +118,21 @@ void LogWriter::inspectMethod(const method_id methodId,
     frameLookup_(frame, jvmti_, *this);
 }
 
-void LogWriter::recordTraceStart(const jint numFrames, const map::HashType threadId) {
+void LogWriter::recordTraceStart(const jint numFrames, map::HashType envHash, const timespec &ts, ThreadBucket *info) {
+    map::HashType threadId = envHash;
+    char *threadName = (char*)"";
+
+    if (info) {
+        threadId = (map::HashType) info->tid;
+        threadName = info->name;
+    }
+
     output_.put(TRACE_START);
     writeValue(numFrames);
     writeValue(threadId);
+    writeValue((int64_t)ts.tv_sec);
+    writeValue((int64_t)ts.tv_nsec);
+    writeWithSize(threadName);
```

Comment:
This implies we record thread name on each stack trace? I would suggest a scheme closer to how we log method names where we have a thread name/id mapping record and let clients sort it out in the view.